src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -35,8 +34,8 @@ def initialize(server, port, options = {})
 
     @server = server
     @port = port
-    @use_ssl = options[:use_ssl]
-    @protocol = @use_ssl ? "https" : "http"
+    self.class.use_ssl = options[:use_ssl]
```

Comment:
Would it make sense to just create the underlying Apache client with the SSL context always and just have "use_ssl" be used at the instance-level -- and not at the class-level at all -- to decide how to set the protocol on each client instance?  I think the SSL context would just be unused for plain-text "http" calls, which wouldn't seem to hurt anything.